synapse/events/__init__.py

```diff
@@ -13,15 +13,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import os
+
 import six
 
 from synapse.util.caches import intern_dict
 from synapse.util.frozenutils import freeze
 
 # Whether we should use frozen_dict in FrozenEvent. Using frozen_dicts prevents
-# bugs where we accidentally share e.g. signature dicts. However, converting
-# a dict to frozen_dicts is expensive.
-USE_FROZEN_DICTS = True
+# bugs where we accidentally share e.g. signature dicts. However, converting a
+# dict to frozen_dicts is expensive.
+#
+# NOTE: This is overridden by the configuration by the Synapse script entry
+# points, but for the sake of tests, it is set here while it cannot be
+# configured on the homeserver object itself.
+USE_FROZEN_DICTS = os.environ.get("SYNAPSE_USE_FROZEN_DICTS", False)
```

Comment:
it feels odd that this will be a (boolean) False if the env var is unset, but a `str` (or a bytes?) if it is set. How about

```
USE_FROZEN_DICTS = "SYNAPSE_USE_FROZEN_DICTS" in os.environ
```
?

Reply:
> Because then SYNAPSE_USE_FROZEN_DICTS=0 would enable them.

that was true with your version too ;)